hw3/hw3-react/src/navigation/AppRouter.js

```diff
@@ -0,0 +1,20 @@
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+
+import HomePage from '../home/HomePage';
+import SearchPage from '../search/SearchPage';
+import HousesPage from '../houses/HousesPage';
+
+function AppRouter() {
+  return (
```

Comment:
Making the API call and using a useState hook in this function would be ideal because once you store the API response data (list of objects), you can pass it to the different components in your Route when it is rendered. For example, if the user hits the /search endpoint, rendering the SearchPage component would look like `<Route path="/houses" element={<SearchPage character={data} />}></Route>`

Reply:
Refactored